lib/util/string-functions.js

```diff
@@ -114,3 +114,11 @@ exports.wordMatch = function wordMatch (string1, string2) {
     [...words1].filter(x => words2.has(x)))
   return intersection.size / Math.max(words1.size, words2.size)
 }
+
+exports.removePageNumber = (pageNum, string) => {
+  // eslint-disable-next-line eqeqeq
+  if (pageNum == string) {
```

Comment:
Use a stronger type check please, convert one to the other as necessary

Reply:
@jenlky There should be more robust pattern matching for a variety of ways to express page number. 

Example:
```
Page 2 of 8
```

Reply:
The number in the page number text may also be different from the index number, depending on where the counting starts. 

Some documents have an unnumbered Cover Page which is essentially Page 0. 
This will push back the index of all subsequent pages.

Reply:
@xumx Yes, I have ran into similar problems with the other pdfs I converted. For example, a page number can look like "4 | Chapter 1: Primer on Latency and Bandwidth" or " The Many Components of Latency | 5 ". 

There could be many possibilities depending on what you convert - office documents, ebooks, letter, etc. I don't know what exactly the owner of the repo wants. 

Thus, I decided to limit the scope of pattern matching to this particular use case - by referring to the pdf in examples/ExamplePdf.pdf. In which case it is a simple page number at the bottom of the document that is also the last element.

Reply:
If so, please leave commentary to that effect